modules/aerodyn/src/AeroDyn.f90

```diff
@@ -1426,7 +1444,9 @@ subroutine SetBuoyancyParameters( InputFileData, u, p, ErrStat, ErrMsg )
          else
             p%BlAxCent(j,k) = ( p%BlRad(j,k)**2 + 2.0_ReKi*p%BlRad(j,k)*p%BlRad(j+1,k) + 3.0_ReKi*p%BlRad(j+1,k)**2 ) / ( 4.0_ReKi*( p%BlRad(j,k)**2 + p%BlRad(j,k)*p%BlRad(j+1,k) + p%BlRad(j+1,k)**2) ) ! fractional axial centroid of element j
          end if
+         tempVolBl(k) = tempVolBl(k) + pi/3.0_ReKi * ( p%BlRad(j,k)**2 + p%BlRad(j,k)*p%BlRad(j+1,k) + p%BlRad(j+1,k)**2 ) * p%BlDL(j,k)
```

Comment:
I'm guessing this is in the implementation plan. Maybe it can be included in the openfast documentation (if not already there)? It looks like an important calculation.

Reply:
The blade and tower buoyancy is not calculated from the volume, and these were added just for the summary file so we could output the rotor buoyancy volume to help with initial platform design. But I will add the equations to the documentation, with some context.